docs/migration.md

```diff
@@ -25,6 +25,35 @@ to restore old behavior.
 
 **Error indications:** Implicit conversion from `uint_least8_t` to `uint8_t`.
 
+### Migrate to bzlmod
+
+**Rationale:** Due to the [shortcomings of the WORKSPACE system](https://bazel.build/external/overview#workspace-shortcomings),
+Bzlmod is going to replace the legacy WORKSPACE system in future Bazel releases.
+Therefore, nanopb has been migrated to use bzlmod to better support newer bazel versions.
+
+**Changes**
+* upgrade bazel deps
+  * rules_python: 0.31.0
+  * rules_proto: 6.0.0-rc2
+  * protobuf: 23.1
+  * rules_proto_grpc: 4.6.0
+* replace WORKSPACE by MODULE.bazel (bzlmod)
+
+**Required actions:** Change the way to import nanopb module in your project
+```py
+bazel_dep(name = "nanopb")
+git_override(
+    module_name = "nanopb",
+    remote = "https://github.com/nanopb/nanopb.git",
+    commit = "<commit>",
+)
+```
+noted that the name of the module has been changed to `nanopb`, to better fit the convention of bzlmod.
+If the old name `com_google_protobuf` is preferred, can add `repo_name` parameter to indicate the repo name.
```

Comment:
Isn't the old name `com_github_nanopb_nanopb`?

Reply:
mb, done